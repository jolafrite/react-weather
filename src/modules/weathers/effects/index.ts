export * from './load-weather/load-weather.effect';
export * from './search-weather/search-weather.effect';
