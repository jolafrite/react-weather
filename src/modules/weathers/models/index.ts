export * from './weather.fixtures';
export * from './model';
