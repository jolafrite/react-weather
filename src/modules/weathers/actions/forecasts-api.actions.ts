import { ActionType } from "../../../common/utils/redux";
import { IWeatherForecast } from "../models";

export const LOAD_ONE_SUCCESS = '[Forecast API] Load Forecast Success';

export const loadOneSuccess = (forecast: IWeatherForecast) => {
  return  {
    type: LOAD_ONE_SUCCESS,
    forecast
  } as const;
};

const all = {
  loadOneSuccess,
};

export type ForecastApiActionsUnion = ActionType<typeof all>;
