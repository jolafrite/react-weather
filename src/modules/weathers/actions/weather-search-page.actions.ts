import { IWeather } from "../models";
import { ActionType } from "../../../common/utils/redux";

export const RESET = '[Weather Search Page] Reset Search Weather';
export const SEARCH = '[Weather Search Page] Search Weather';
export const SEARCH_SUCCESS = '[Weather Search Page] Search Weathers Success';
export const SEARCH_ERROR = '[Weather Search Page] Search Weathers Error';

export const reset = () => {
  return {
    type: RESET,
  } as const;
};

export const search = (query: string) => {
  return {
    type: SEARCH,
    query,
  } as const;
};

export const searchSuccess = (weathers: IWeather[]) => {
  return {
    type: SEARCH_SUCCESS,
    weathers,
  } as const;
};

export const searchError = (error: Error) => {
  return {
    type: SEARCH_ERROR,
    error
  } as const;
};

const all = {
  reset,
  search,
  searchSuccess,
  searchError,
};

export type WeatherSearchPageActionsUnion = ActionType<typeof all>;
