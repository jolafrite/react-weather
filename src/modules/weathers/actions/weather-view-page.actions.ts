import { ActionType } from "../../../common/utils/redux";
import { IWeather, IWeatherForecast } from "../models";

export const SELECT_ONE = '[Weather View Page] Select Weather';
export const LOAD_ONE = '[Weather View Page] Load Weather';
export const LOAD_ONE_SUCCESS = '[Weather View Page] Load Weather Success';
export const LOAD_ONE_ERROR = '[Weather View Page] Load Weather Error';

export const selectOne = (id: number) => {
  return {
    type: SELECT_ONE,
    id,
  } as const;
};

export const loadOne = () => {
  return {
    type: LOAD_ONE,
  } as const;
};

export const loadOneSuccess = (weather: IWeather, forecasts?: IWeatherForecast[]) => {
  return {
    type: LOAD_ONE_SUCCESS,
    weather,
    forecasts,
  } as const;
};

export const loadOneError = (error: Error) => {
  return {
    type: LOAD_ONE_ERROR,
    error
  } as const;
};

const all = {
  selectOne,
  loadOne,
  loadOneSuccess,
  loadOneError,
};

export type WeatherViewPageActionsUnion = ActionType<typeof all>;
