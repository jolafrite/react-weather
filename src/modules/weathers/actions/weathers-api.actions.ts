import { IWeather } from "../models";
import { ActionType } from "../../../common/utils/redux";

export const LOAD_ONE_SUCCESS = '[Weather API] Load Weather Success';
export const LOAD_LIST_SUCCESS = '[Weather API] Load Weathers Success';

export const loadOneSuccess = (weather: IWeather) => {
  return {
    type: LOAD_ONE_SUCCESS,
    weather,
  } as const;
};

export const loadListSuccess = (weathers: IWeather[]) => {
  return  {
    type: LOAD_LIST_SUCCESS,
    weathers
  } as const;
};

const all = {
  loadOneSuccess,
  loadListSuccess,
};

export type WeatherApiActionsUnion = ActionType<typeof all>;
