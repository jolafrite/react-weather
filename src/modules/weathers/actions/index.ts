import * as WeatherSearchPageActions from './weather-search-page.actions';
import * as WeatherViewPageActions from './weather-view-page.actions';
import * as WeatherApiActions from './weathers-api.actions';

export {
  WeatherSearchPageActions,
  WeatherViewPageActions,
  WeatherApiActions,
};
