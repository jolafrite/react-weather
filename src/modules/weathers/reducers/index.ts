import { combineReducers } from "redux";
import * as ForWeatherSearchPage from './weather-search-page.reducer';
import * as ForWeatherViewPage from './weather-view-page.reducer';
import * as ForWeathers from './weathers.reducer';

export type State = {
  readonly weathers: ForWeathers.State,
  readonly searchPage: ForWeatherSearchPage.State,
  readonly viewPage: ForWeatherViewPage.State,
};

export const reducer = combineReducers<State>({
  weathers: ForWeathers.reducer,
  searchPage: ForWeatherSearchPage.reducer,
  viewPage: ForWeatherViewPage.reducer,
});
