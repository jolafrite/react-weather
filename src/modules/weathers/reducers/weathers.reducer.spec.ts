import { reducer, State } from "./weathers.reducer";
import { WeatherApiActions } from "../actions";
import { weatherFixtures } from "../models";

describe('Weathers reducer', () => {

  describe('action: undefined', () => {
    test('should result the default state', () => {
      expect(
        reducer(undefined, {} as any)
      ).toMatchSnapshot();
    });
  });

  describe('action: WeatherApiActions.LoadListSuccess', () => {
    test('should save the weathers', () => {
      const weathers = [
        weatherFixtures({ id: 1 }),
        weatherFixtures({ id: 2 }),
      ];

      expect(
        reducer(undefined, WeatherApiActions.loadListSuccess(weathers))
      ).toMatchSnapshot();
    });

    test('should replace the existing weathers', () => {
      const state: State = {
        entries: {
          1: weatherFixtures({ id: 1 }),
        }
      };
      const weathers = [
        weatherFixtures({ id: 1, name: 'A new name' }),
      ];

      expect(
        reducer(state, WeatherApiActions.loadListSuccess(weathers))
      ).toMatchSnapshot();
    });
  });

  describe('action: WeatherApiActions.LoadOneSuccess', () => {
    test('should save the weather', () => {
      const weather = weatherFixtures({ id: 1 });

      expect(
        reducer(undefined, WeatherApiActions.loadOneSuccess(weather))
      ).toMatchSnapshot();
    });
  });

});

