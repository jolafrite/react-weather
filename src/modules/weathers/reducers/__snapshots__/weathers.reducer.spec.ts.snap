// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weathers reducer action: WeatherApiActions.LoadListSuccess should replace the existing weathers 1`] = `
Object {
  "entries": Object {
    "1": Object {
      "city": Object {
        "coord": Object {
          "lat": 34.966671,
          "lon": 138.933334,
        },
        "country": "JP",
        "id": 1851632,
        "name": "Shuzenji",
      },
      "cnt": 38,
      "id": 1,
      "list": Array [
        Object {
          "clouds": Object {
            "all": 88,
          },
          "dt": 1406106000,
          "dt_txt": "2014-07-23 09:00:00",
          "main": Object {
            "grnd_level": 1005.93,
            "humidity": 87,
            "pressure": 1005.93,
            "sea_level": 1018.18,
            "temp": 298.77,
            "temp_kf": 0.26,
            "temp_max": 298.774,
            "temp_min": 298.77,
          },
          "weather": Array [
            Object {
              "description": "overcast clouds",
              "icon": "04d",
              "id": 804,
              "main": "Clouds",
            },
          ],
          "wind": Object {
            "deg": 229.501,
            "speed": 5.71,
          },
        },
      ],
    },
    "1851632": Object {
      "city": Object {
        "coord": Object {
          "lat": 34.966671,
          "lon": 138.933334,
        },
        "country": "JP",
        "id": 1851632,
        "name": "Shuzenji",
      },
      "cnt": 38,
      "id": 1,
      "list": Array [
        Object {
          "clouds": Object {
            "all": 88,
          },
          "dt": 1406106000,
          "dt_txt": "2014-07-23 09:00:00",
          "main": Object {
            "grnd_level": 1005.93,
            "humidity": 87,
            "pressure": 1005.93,
            "sea_level": 1018.18,
            "temp": 298.77,
            "temp_kf": 0.26,
            "temp_max": 298.774,
            "temp_min": 298.77,
          },
          "weather": Array [
            Object {
              "description": "overcast clouds",
              "icon": "04d",
              "id": 804,
              "main": "Clouds",
            },
          ],
          "wind": Object {
            "deg": 229.501,
            "speed": 5.71,
          },
        },
      ],
      "name": "A new name",
    },
  },
}
`;

exports[`Weathers reducer action: WeatherApiActions.LoadListSuccess should save the weathers 1`] = `
Object {
  "entries": Object {
    "1851632": Object {
      "city": Object {
        "coord": Object {
          "lat": 34.966671,
          "lon": 138.933334,
        },
        "country": "JP",
        "id": 1851632,
        "name": "Shuzenji",
      },
      "cnt": 38,
      "id": 2,
      "list": Array [
        Object {
          "clouds": Object {
            "all": 88,
          },
          "dt": 1406106000,
          "dt_txt": "2014-07-23 09:00:00",
          "main": Object {
            "grnd_level": 1005.93,
            "humidity": 87,
            "pressure": 1005.93,
            "sea_level": 1018.18,
            "temp": 298.77,
            "temp_kf": 0.26,
            "temp_max": 298.774,
            "temp_min": 298.77,
          },
          "weather": Array [
            Object {
              "description": "overcast clouds",
              "icon": "04d",
              "id": 804,
              "main": "Clouds",
            },
          ],
          "wind": Object {
            "deg": 229.501,
            "speed": 5.71,
          },
        },
      ],
    },
  },
}
`;

exports[`Weathers reducer action: WeatherApiActions.LoadOneSuccess should save the weather 1`] = `
Object {
  "entries": Object {
    "1851632": Object {
      "city": Object {
        "coord": Object {
          "lat": 34.966671,
          "lon": 138.933334,
        },
        "country": "JP",
        "id": 1851632,
        "name": "Shuzenji",
      },
      "cnt": 38,
      "id": 1,
      "list": Array [
        Object {
          "clouds": Object {
            "all": 88,
          },
          "dt": 1406106000,
          "dt_txt": "2014-07-23 09:00:00",
          "main": Object {
            "grnd_level": 1005.93,
            "humidity": 87,
            "pressure": 1005.93,
            "sea_level": 1018.18,
            "temp": 298.77,
            "temp_kf": 0.26,
            "temp_max": 298.774,
            "temp_min": 298.77,
          },
          "weather": Array [
            Object {
              "description": "overcast clouds",
              "icon": "04d",
              "id": 804,
              "main": "Clouds",
            },
          ],
          "wind": Object {
            "deg": 229.501,
            "speed": 5.71,
          },
        },
      ],
    },
  },
}
`;

exports[`Weathers reducer action: undefined should result the default state 1`] = `
Object {
  "entries": Object {},
}
`;
